doc/sphinx/practical-tools/coq-commands.rst

```diff
@@ -56,14 +56,17 @@ By resource file
 ~~~~~~~~~~~~~~~~~~~~~~~
 
 When |Coq| is launched, with either ``coqtop`` or ``coqc``, the resource file
-``$XDG_CONFIG_HOME/coq/coqrc.xxx`` is loaded, where ``$XDG_CONFIG_HOME``
+``$XDG_CONFIG_HOME/coq/coqrc.xxx`` is loaded, if it exists. Here, ``$XDG_CONFIG_HOME``
 is the configuration directory of the user (by default its home
-directory ``/.config`` and ``xxx`` is the version number (e.g. 8.8). If
+directory ``~/.config``) and ``xxx`` is the version number (e.g. 8.8). If
 this file is not found, then the file ``$XDG_CONFIG_HOME/coqrc`` is
-searched. You can also specify an arbitrary name for the resource file
+searched. If not found, it is the file ``~/.coqrc.xxx`` which is searched,
+and, if still not found, the file ``~/.coqrc``. If the latter is also
+absent, no resource file is loaded.
+You can also specify an arbitrary name for the resource file
 (see option ``-init-file`` below).
 
-This file may contain, for instance, ``Add LoadPath`` commands to add
+The resource file may contain, for instance, ``Add LoadPath`` commands to add
```

Comment:
Isn't this considered bad practice?

Reply:
It is a bit archane but not a bad practice per se, it is equivalente to setting some global -R / -Q options.

Reply:
Isn't it basically the only point of the existence of this file anyways?

Reply:
@herbelin you may want to be more explicit and reflect the current semantics: "the content of the resource file will be appended to the beginning of each document read by Coq".

Reply:
But it is not included in vo files, isn't it?

Reply:
Yup, I think it is included.

Reply:
Ah, interesting!

But after all, as soon as we accept any command there, we have to store the potential dependencies somewhere, and there is no other place for that than the .vo file.

I changed into: "When Coq is launched, with either coqtop or coqc, the
resource file $XDG_CONFIG_HOME/coq/coqrc.xxx, if it exists, will
be implicitly prepended to any document read by Coq, whether it is an
interactive session or a file to compile.". Is that ok?

Reply:
Yup, that's seems a bit clearer about the purpose, thanks. It is hard to come up with good wording for this kind of stuff.

As far as I can trace back in history the commands in the file were always processed "normally", as @Zimmi48 points out indeed the utility of such rc file in the current form is a bit questionable, but it doesn't create too much of a problem I'd say. For libraries we want to provide higher-level primitives anyways.